analysis/bin/vtsAnalyser.cc

```diff
@@ -562,154 +573,55 @@ int vtsAnalyser::Test() {
-          b_hadTruth_dr_jetMat_vec.push_back(jet_tlv.DeltaR(had_tlv));
-          b_hadTruth_pt_jetMat_vec.push_back(had_pt[i]);
-          b_hadTruth_eta_jetMat_vec.push_back(had_eta[i]);
-          b_hadTruth_phi_jetMat_vec.push_back(had_phi[i]);
-          b_hadTruth_mass_jetMat_vec.push_back(had_mass[i]);
-          b_hadTruth_lxy_jetMat_vec.push_back(had_lxy[i]);
-          b_hadTruth_lxySig_jetMat_vec.push_back(had_lxy[i]/had_lxyErr[i]);
-          b_hadTruth_angleXY_jetMat_vec.push_back(had_angleXY[i]);
-          b_hadTruth_angleXYZ_jetMat_vec.push_back(had_angleXYZ[i]);
-          b_hadTruth_chi2_jetMat_vec.push_back(had_chi2[i]);
-          b_hadTruth_dca_jetMat_vec.push_back(had_dca[i]);
-          b_hadTruth_l3D_jetMat_vec.push_back(had_l3D[i]);
-          b_hadTruth_l3DSig_jetMat_vec.push_back(had_l3D[i]/had_l3DErr[i]);
-          b_hadTruth_legDR_jetMat_vec.push_back(had_legDR[i]);
-          b_hadTruth_pdgId_jetMat_vec.push_back(had_pdgId[i]);
-          b_hadTruth_dau1_chi2_jetMat_vec.push_back(had_dau1_chi2[i]);
-          b_hadTruth_dau1_ipsigXY_jetMat_vec.push_back(had_dau1_ipsigXY[i]);
-          b_hadTruth_dau1_ipsigZ_jetMat_vec.push_back(had_dau1_ipsigZ[i]);
-          b_hadTruth_dau1_pt_jetMat_vec.push_back(had_dau1_pt[i]);
-          b_hadTruth_dau2_chi2_jetMat_vec.push_back(had_dau2_chi2[i]);
-          b_hadTruth_dau2_ipsigXY_jetMat_vec.push_back(had_dau1_ipsigXY[i]);
-          b_hadTruth_dau2_ipsigZ_jetMat_vec.push_back(had_dau1_ipsigZ[i]);
-          b_hadTruth_dau2_pt_jetMat_vec.push_back(had_dau1_pt[i]);
-        } 
-      }
-      // KS with cuts from HadronAnalysis()
-      if (had_pdgId[i] == 310) {
-        if ( (had_lxy[i]/had_lxyErr[i]) < 3 ) continue;
-        if ( had_angleXY[i] < 0.98 ) continue;
-        if ( had_chi2[i] > 3 ) continue;
-        if ( had_dca[i] > 1 ) continue;
-        if ( (jet_tlv.DeltaR(had_tlv) < 0.5) && (had_pt[i]/Jet_pt[j] > 0.15)) {
-          ++nKSWithCut;
-          b_had_isFrom_vec.push_back(hadTruth_isHadFromTsb[i]);
-          b_had_qjMapForMC_vec.push_back(qjMapForMC_[j]);
-          b_had_x_vec.push_back(had_pt[i]/Jet_pt[j]);
-          b_had_d_vec.push_back(GetD(had_pt[i], had_eta[i], had_phi[i], had_mass[i], had_x[i], had_y[i], had_z[i]));
-          b_had_dr_vec.push_back(jet_tlv.DeltaR(had_tlv));
-          if ( had_dau1_pt[i] >= 0.95 && had_dau2_pt[i] >= 0.95) {
-            ++nKSWithDauCut;
-            b_had_isFrom_dc_vec.push_back(hadTruth_isHadFromTsb[i]);
-            b_had_qjMapForMC_dc_vec.push_back(qjMapForMC_[j]);
-            b_had_x_dc_vec.push_back(had_pt[i]/Jet_pt[j]);
-            b_had_d_dc_vec.push_back(GetD(had_pt[i], had_eta[i], had_phi[i], had_mass[i], had_x[i], had_y[i], had_z[i]));
-            b_had_dr_dc_vec.push_back(jet_tlv.DeltaR(had_tlv));
-          } 
-        }
-      } 
+      auto dr = jet_tlv.DeltaR(had_tlv);
+      if (dr < drReco ) {
+        drReco = jet_tlv.DeltaR(had_tlv);
```

Comment:
Same comment as before, you can just use the calculated value

Reply:
yep